### DISCUSSION TITLE
graphql api resource query is case sensitive

### DISCUSSION THREAD

**Discussion by myrfy001** [community/community#21802] [New to GitHub]
<p>Hi</p>
<p>The graphql api resource query is case sensitive. When I try to query a object using url, for example:</p>
<p><a href="https://github.com/by2hit" rel="nofollow noopener">https://github.com/by2hit</a> will not work, but <a href="https://github.com/BY2HIT" rel="nofollow noopener">https://github.com/BY2HIT</a> is working.</p>
<p>this will not work:</p>
<pre><code>{
    resource(url: "https://github.com/by2hit") {
    __typename
}
</code></pre>
<p>this will work:</p>
<pre><code>{
    resource(url: "https://github.com/BY2HIT") {
    __typename
}
</code></pre>
<p>On the other hand, type both the lowercased or the upper cased url in a browser can visit the same page.</p>

**Comment by li-pamela** [comment]
<p>Hi <a class="mention" href="/myrfy001">@myrfy001</a>,</p>
<p>This post was moved to a different board that fits your topic of discussion a bit better. This means you’ll get better engagement on your post, and it keeps our Community organized so users can more easily find information.</p>
<p>As you’ll notice, your Topic is now in the Project Development Help board. No action is needed on your part; you can continue the conversation as normal here.</p>
<p>Let me know if you have any other questions or if I can help with anything else.</p>
<p>Cheers!</p>

**Comment by ThirdEyeRose** [comment]
<p>Hi <a class="mention" href="/myrfy001">@myrfy001</a>,</p>
<p>Our apologies, we incorrectly moved this to the Project Development Help and Advice Board when it should be in the How to use Git and GitHub forum since I believe you are reporting issues with <a href="https://developer.github.com/v4/" rel="nofollow noopener">GitHub’s GraphQL API</a>. I have moved this post to the How to use Git and GitHub board where you can continue this discussion further.</p>
<p>Thanks!</p>

**Comment by schunlee** [comment]
<p>where to find the topic been moved by you. how to make case-insensitive query?</p>

**Comment by mpboom** [comment, marked as answer]
<p>This may be due to the fact that the API uses different systems than the browser. It’s best to just keep it case-sensitive on your end - nothing can go wrong then.</p>

**Comment by barnesdc** [comment]
<p>Is there any updates to this. Specifically if I want to query a file, is there a way to make that query case-insensitive. For example shown below I am querying multiple project’s README.md files, but this will not work correctly when the files are named differently (e.g. Readme.md or readme.md or ReadMe.md, etc)</p>
<pre><code>readme: object(expression:"master:README.md") { ... on Blob {
  text
  }
}
</code></pre>
